R/loo_difference_plot.R

```diff
@@ -0,0 +1,151 @@
+#' Compare models across domains
+#' 
+#' The LOO difference plot shows how the ELPD of two different models
+#' changes when a predictor is varied. This can is useful for identifying
+#' opportunities for model stacking or expansion.
+#' 
+#' @param y A vector of observations. See Details.
+#' @param psis_object_1,psis_object_2 If using loo version 2.0.0 or greater, 
+#' an object returned by the `[loo::psis()]` function (or by the 
+#' `[loo::loo()]` function with argument `save_psis` set to `TRUE`).
+#' @param ... Currently unused.
+#' @param group A grouping variable (a vector or factor) the same length 
+#' as `y`. Each value in group is interpreted as the group level pertaining 
+#' to the corresponding value of `y`. If `FALSE`, ignored.
+#' @param size,alpha,jitter Passed to `[ggplot2::geom_point()]` to control 
+#' aesthetics. `size` and `alpha` are passed to to the `size` and `alpha` 
+#' arguments of `[ggplot2::geom_jitter()]` to control the appearance of 
+#' points. `jitter` can be either a number or a vector of numbers.
+#' Passing a single number will jitter variables along the x axis only, while 
+#' passing a vector will jitter along both axes.
+#' @param quantiles Boolean that determines whether to plot the quantiles of
+#' `y` rather than `y` itself. Useful when `y` has a very irregular 
+#' distribution.
+#' @param sortByGroup Sort observations by `group`, then plot against an 
+#' arbitrary index. Plotting by index can be useful when categories have 
+#' very different sample sizes. 
+#' 
+#' 
+#' @template return-ggplot
+#' 
+#' @template reference-vis-paper
+#' 
+#' @examples 
+#' 
+#' library(loo)
+#' 
+#' cbPalette <- c("#636363", "#E69F00", "#56B4E9", "#009E73", 
+#'                "#F0E442", "#0072B2","#CC79A7")
+#' 
+#' # Plot using groups from WHO
+#' 
+#' plot_loo_dif(factor(GM@data$super_region_name), loo3, loo2, 
+#'              group = GM@data$super_region_name, alpha = .5, 
+#'              jitter = c(.45, .2)
+#'              ) + 
+#'              xlab("Region") + scale_colour_manual(values=cbPalette)
+#' 
+#' # Plot using groups identified with clustering
+#' 
+#' plot_loo_dif(factor(GM@data$cluster_region), loo3, loo2, 
+#'              group = GM@data$super_region_name, alpha = .5, 
+#'              jitter = c(.45, .2)
+#'              ) + 
+#'              xlab("Cluster Group") + scale_colour_manual(values=cbPalette)
+#'              
+#' # Plot using an index variable to reduce crowding
+#' 
+#' plot_loo_dif(1:2980, loo3, loo2, group = GM@data$super_region_name, 
+#'              alpha = .5, sortByGroup = TRUE, 
+#'              ) + 
+#'              xlab("Index") + scale_colour_manual(values=cbPalette)
+#' 
+#' 
+plot_loo_variation <- 
+  function(y,
+           psis_object_1,
+           psis_object_2,
+           ...,
+           group = FALSE,
```

Comment:
I would use `NULL` instead of `FALSE`. To me a default of `FALSE` gives the impression that the possible values are TRUE and FALSE, but `group=TRUE` isn't a possible value. So I think `NULL` makes more sense as the default. Same with `outlier_thresh`.   On the other hand, `FALSE` makes sense for `quantiles` since it seems to actually be an argument that needs a logical/boolean value.